pallets/pool-xyk/src/math.rs

```diff
@@ -200,13 +204,25 @@ impl<T: Config> Pallet<T> {
             // x_in * (1 - fee) = (x * y_out) / (y - y_out)
             let fxw_y_out = fxw_y_out.clone() + Fixed::from_bits(1); // by 1 correction to overestimate required input
             let nominator = fxw_x * fxw_y_out.clone();
+            ensure!(
+                to_balance!(fxw_y.clone()) > to_balance!(fxw_y_out.clone()),
+                Error::<T>::NotEnoughLiquidityForSwap
+            );
             let denominator = fxw_y - fxw_y_out;
             let x_in_without_fee = nominator / denominator;
             let x_in = if deduce_fee {
                 x_in_without_fee.clone() / (fixed_wrapper!(1) - fee_fraction)
             } else {
                 x_in_without_fee.clone()
             };
+            ensure!(
+                x_in.clone().get().is_ok(),
+                Error::<T>::NotEnoughLiquidityForSwap
+            );
+            ensure!(
+                x_in >= x_in_without_fee,
+                Error::<T>::NotEnoughLiquidityForSwap
+            );
```

Comment:
seems to be possible only if `fee_fraction` is negative (or some overflow happened, which should be caught by `ensure` right above). does this check makes sense?